DEVELOPER.md

```diff
@@ -26,8 +26,38 @@ Envoy Gateway is built using a [make][make]-based build system. Our CI is based
   separately.
 
 ## Quick start
+* Run `make help` to see all the available targets to build, test and run `envoy-gateway`.
 
-Run `make help` to see all the available targets to build, test and run `envoy-gateway`.
+### Building the `envoy-gateway` binary
+* Run `make build` to build the binary that gets generated in the `bin/` directory
+
+### Running tests
+* Run `make test` to run the golang tests.
+
+### Running code linters
+* Run `make lint` to make sure your code passes all the linter checks.
+
+### Building and Pushing the Image
+* Run `IMAGE=foo/bar make image` to build the docker image.
+* Run `IMAGE=foo/bar make push-multiarch` to build and push the multi-arch docker image.
+
+**_NOTE:_**  Replace `IMAGE` with your registry's image name.
```

Comment:
Can you be more specific with this note, and/or provide an example above? `IMAGE` should be `REGISTRY/REPOSITORY/IMAGE_NAME`. For example:
```
IMAGE=docker.io/you/gateway-dev make push-multiarch
```